Source/WebGPU/WebGPU/RenderBundleEncoder.mm

```diff
@@ -749,7 +753,8 @@ - (instancetype)initWithICB:(id<MTLIndirectCommandBuffer>)icb containerBuffer:(i
 
     auto indexCountTimesSizeInBytes = checkedProduct<size_t>(indexCount, indexSizeInBytes);
     if (id<MTLIndirectRenderCommand> icbCommand = currentRenderCommand()) {
-        storeVertexBufferCountsForValidation(indexCount, instanceCount, indexBufferOffsetInBytes / indexSizeInBytes, baseVertex, firstInstance, m_indexType, indexBufferOffsetInBytes);
+        if (!storeVertexBufferCountsForValidation(indexCount, instanceCount, indexBufferOffsetInBytes / indexSizeInBytes, baseVertex, firstInstance, m_indexType, indexBufferOffsetInBytes))
+            return finalizeRenderCommand();
```

Comment:
Do we need to makeInvalid in here?

Reply:
I don’t think so because this is due to a very confusing (to me at least) part of the specification where ‘destroyed objects should continue to function like normal objects’, until they are submitted as part of a GPUCommandBuffer

Reply:
See https://gpuweb.github.io/gpuweb/explainer/#early-destroy

> Unlike invalidity, destroyed-ness can change after creation, is not contagious, and is validated only when work is actually submitted (e.g. queue.writeBuffer() or queue.submit()), not when creating dependent objects (like command encoders, see above).

we had initially not assumed this 2 years ago and all the related tests failed

Reply:
Likely the motivation is while the index buffer is used as part of the draw call with Metal, on another backend, it is only used in the setIndexBuffer call

Reply:
We handle the make the command buffer commit invalid via setCommandEncoder on the index buffer